system/ee/ExpressionEngine/Model/Channel/ChannelEntry.php

```diff
@@ -1108,8 +1108,10 @@ public function populateChannels($field)
 
 		foreach ($channels as $channel)
 		{
+			$channel_statuses = $channel->Statuses->getIds();
+
 			if ($my_fields == $channel->getAllCustomFields()->pluck('field_id') &&
```

Comment:
Any idea why we need `$my_fields` here? It does not seem necessary here, but maybe I'm, missing something

Reply:
It's what it was named, so I left it.  But basically, it's the current channel's fields.  So you're comparing the current ($my_fields and $my_statuses) fields and statuses to the fields and statuses of every channel.  And only ones that match on both the fields and the statuses go into the channel dropdown.

The goobered up order had it so even the current channel's statuses weren't matching that channel's statuses when retrieved via $channel->Statuses->getIds(); so there were no options in the channel select.  And it's required, so you couldn't even save.  

But the names $my_fields and $my_statuses were already there, so I just left them.  I think in this case 'my' is 'the current entry's channel'.

Reply:
Ok, I thought you might know - we're probably fine leaving it there now, though this seems like some place for optimization